### PR TITLE
[WIP] MSan: also compile deps with -fsanitize=memory

### PR DESCRIPTION
**Pull request by justinmk**
https://github.com/neovim/neovim/issues/2957#issuecomment-129651985

### CODE REVIEW

File: third-party/cmake/BuildLibuv.cmake [hunk 1]
```diff
@@ -41,18 +41,22 @@ set(UNIX_CFGCMD sh ${DEPS_BUILD_DIR}/src/libuv/autogen.sh &&
...
 if(UNIX)
-  BuildLibUv(
+  BuildLibuv(
     CONFIGURE_COMMAND ${UNIX_CFGCMD}
-    INSTALL_COMMAND ${MAKE_PRG} install)
+    INSTALL_COMMAND ${MAKE_PRG} install
+                                CFLAGS+=-fno-omit-frame-pointer
+                                CFLAGS+=-fno-optimize-sibling-calls
+                                CFLAGS+=-fsanitize-memory-track-origins
+                                CFLAGS+=-fsanitize=memory)
```
Comment (oni-link) [inline third-party/cmake/BuildLibuv.cmake]: gcc does not know the last two options.

Comment (justinmk) [inline third-party/cmake/BuildLibuv.cmake]: :+1:  currently this is just an experiment to see if the false-positive will go away.

**Comment by justinmk** [comment]
```
0m[31m[1mLinking C shared module ../../lib/libnvim-test.so
[0m/home/travis/nvim-deps/usr/lib/libuv.a(libuv_la-uv-common.o): In function `uv__strdup':
/home/travis/nvim-deps/build/src/libuv/src/uv-common.c:(.text+0x2d): undefined reference to `__msan_param_tls'
/home/travis/nvim-deps/build/src/libuv/src/uv-common.c:(.text+0x38): undefined reference to `__msan_param_origin_tls'
/home/travis/nvim-deps/build/src/libuv/src/uv-common.c:(.text+0x8d): undefined reference to `__msan_set_alloca_origin4'
/home/travis/nvim-deps/build/src/libuv/src/uv-common.c:(.text+0xf0): undefined reference to `__msan_set_alloca_origin4'
/home/travis/nvim-deps/build/src/libuv/src/uv-common.c:(.text+0x16b): undefined reference to `__msan_set_alloca_origin4'
```

**Comment by aktau** [comment]
The MSan option should probably be removed (or at least a warning added) from the CMake file until/if this gets fixed, because the same false positive from 2015 is still there.

**Comment by dundargoc** [comment]
> The MSan option should probably be removed (or at least a warning added) from the CMake file until/if this gets fixed, because the same false positive from 2015 is still there.

Would you like to send a PR? I doubt most people have the context of this issue like you do.

**Pull request closed without merging**